RELEASE_NOTES.md

```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+* preserve checkout history so that `git checkout -` works as expected alongside Git Town commands
+  ([#668](https://github.com/Originate/git-town/pull/668))
```

Comment:
All the other links are actually links to issues instead of pull requests. Thus this should be #65.